src/object/srv_mod.c

```diff
@@ -79,16 +77,64 @@ obj_tls_fini(const struct dss_thread_local_storage *dtls,
 	if (tls->ot_echo_sgl.sg_iovs != NULL)
 		daos_sgl_fini(&tls->ot_echo_sgl, true);
 
+	if (tls->ot_sp)
+		srv_profile_destroy(tls->ot_sp);
+
 	D_FREE(tls);
 }
 
+static char *profile_op_names[] = {
+	"update_prep",
+	"update_local",
+	"update_end",
+	"update_repl",
+	"update"
```

Comment:
they should stay together with opcode definitions above, probably define a data structure which has two members: opc and string name, then define an array for all these operations

Reply:
"So we consume a slot to record duration each time, this is not a lot because ideally each ULT can handle 100K+ RPC/sec,"

Oh, this is just the header, and each chunk will record 10K ops, if it uses up, then it do extra allocation. In the mean time, the dump thread will keep dump the profile ops to the dump file, and release these chunks.

Reply:
"they should stay together with opcode definitions above, probably define a data structure which has two members: opc and string name, then define an array for all these operations"